pyccel/codegen/printing/ccode.py

```diff
@@ -586,6 +602,68 @@ def function_signature(self, expr):
         else:
             return '{0}{1}({2})'.format(ret_type, name, arg_code)
 
+    def _print_IndexedElement(self, expr):
+        if isinstance(expr.base, IndexedVariable):
+            base = expr.base.internal_variable
+        else:
+            base = expr.base
+        base = self._print(expr.base.label)
+        inds = list(expr.indices)
+        inds = inds[::-1]
+        base_shape = Shape(expr.base)
+        allow_negative_indexes = (isinstance(expr.base, IndexedVariable) and \
+                expr.base.internal_variable.allows_negative_indexes)
+        for i, ind in enumerate(inds):
+            if isinstance(ind, PyccelUnarySub) and isinstance(ind.args[0], Integer):
+                inds[i] = PyccelMinus(base_shape[i], ind.args[0])
+            else:
+                if isinstance(ind, Slice):
+                    #setting the slice start and end to their correct value if none is provided
+                    start = ind.start
+                    end = ind.end
+                    if ind.start is None:
+                        start = 0
+                    if ind.end is None:
+                        end = expr.base.shape[i]
+                    inds[i] = Slice.__new__(Slice, start, end)
+                #indices of indexedElement of len==1 shouldn't be a Tuple
+                if isinstance(ind, Tuple) and len(ind) == 1:
+                    inds[i].args = ind[0]
+                if allow_negative_indexes and not isinstance(ind, Integer):
+                    inds[i] = PyccelMod(ind, base_shape[i])
+        inds = [self._print(i) for i in inds]
+        #set dtype to the C struct types
+        dtype = self.find_in_dtype_registry(format(expr.dtype), expr.precision, array=True)
+        if expr.rank > 0:
+            return "array_slicing(%s, %s)" % (base, ", ".join(inds))
+        return "%s.nd_%s[get_index(%s, %s)]" % (base, dtype, base, ", ".join(inds))
+
+    def _print_Allocate(self, expr):
+        free_code = ''
+        #free the array if its already allocated
+        if  (expr.status != 'unallocated'):
+            free_code = 'if (%s.raw_data != NULL)\n' % self._print(expr.variable.name)
+            free_code += '{\nfree_array(%s);\n}\n' % self._print(expr.variable.name)
+        self._additional_imports.add('ndarrays')
+        shape = ", ".join(str(a) for a in expr.shape)
+        dtype = self.find_in_dtype_registry(format(expr.variable.dtype), expr.variable.precision, array=True)
+        shape_name = "shape_dummy_%s" % format(expr.__hash__()%991)
```

Comment:
Please use the `parser.get_new_name` function (I might have the wrong method name, it's defined in pyccel.parser.base) to get a new name which avoids name clashes

Reply:
I checked. That is the right function name. You can either use:
```
self.parser.get_new_name()
```
to get a new dummy name. Or you can use the `create_incremented_string` method from `ast.core` to get a similar naming scheme to what you are currently using.
In this case the call should be:
```
create_incremented_string(self.parser.used_names, prefix = 'shape_dummy')
```

Reply:
thank you, will fix it